### PR TITLE
api: support preview mode

### PR DESCRIPTION
**Pull request by khairul-alam-licon**
# Description
- Deprecated the "isTestMode" in `MiniAppSdkConfig`
- Added "isPreviewMode" support in `MiniAppSdkConfig`
- Change the API endpoint for "isPreviewMode" from "test" to "preview"
- Add "isPreviewMode" to user guide and changelog.
- Adopt Preview Mode in sample app. "Preview Mode" switch in sample app settings is visible now.

## Links
- MINI-2141

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

**Review by minh-rakuten — APPROVED** [review]
👍 👍

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,9 +5,11 @@
 **SDK**
 - **Feature:** Mini App can call media execution play/pause programmatically.
 - **Change:** Added the default implementation for external link handler. Using [custom tab](https://developers.google.com/web/android/custom-tabs).
+- **Change:** Adopted Preview Mode to get the miniapps.
```
Comment (corycaywood) [inline CHANGELOG.md:8]: You should also mention that `isTestMode` changed to `isPreviewMode`.

File: miniapp/USERGUIDE.md [hunk 2]
```diff
@@ -58,6 +58,11 @@ In your `AndroidManifest.xml`:
             android:name="com.rakuten.tech.mobile.miniapp.BaseUrl"
             android:value="https://www.example.com" />
 
+        <!-- Preview mode used for retrieving the Mini Apps -->
+        <meta-data
+            android:name="com.rakuten.tech.mobile.miniapp.IsPreviewMode"
```
Comment (corycaywood) [inline miniapp/USERGUIDE.md:63]: This also needs to be added to the table of settings above.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniappSdkInitializer.kt [hunk 3]
```diff
@@ -72,7 +78,7 @@ class MiniappSdkInitializer : ContentProvider() {
                 subscriptionKey = manifestConfig.subscriptionKey(),
                 hostAppVersionId = manifestConfig.hostAppVersion(),
                 hostAppUserAgentInfo = manifestConfig.hostAppUserAgentInfo(),
-                isTestMode = manifestConfig.isTestMode()
+                isTestMode = manifestConfig.isPreviewMode()
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniappSdkInitializer.kt:81]: `isTestMode` is deprecated, but it should still work if they have only set that value and not `isPreviewMode`. So maybe this instead?
```suggestion
                isTestMode = manifestConfig.isPreviewMode() || manifestConfig.isTestMode()
```

File: testapp/build.gradle [hunk 4]
```diff
@@ -35,7 +35,7 @@ android {
     def appName = "Mini App Sample"
     def stagingManifestPlaceholders = [
             baseUrl                 : property("MINIAPP_SERVER_BASE_URL") ?: "https://www.example.com/",
-            isTestMode              : property("IS_TEST_MODE") ?: false,
+            isPreviewMode           : property("IS_PREVIEW_MODE") ?: false,
```
Comment (corycaywood) [inline testapp/build.gradle:38]: This should be `true` by default for the demo app.

File: testapp/build.gradle [hunk 5]
```diff
@@ -44,7 +44,7 @@ android {
     ]
     def prodManifestPlaceholders = [
             baseUrl                 : property("MINIAPP_PROD_SERVER_BASE_URL") ?: "https://www.example.com/",
-            isTestMode              : property("PROD_IS_TEST_MODE") ?: false,
+            isPreviewMode           : property("PROD_IS_PREVIEW_MODE") ?: false,
```
Comment (corycaywood) [inline testapp/build.gradle:47]: Same here, this should be `true` by default.

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/AppSettings.kt [hunk 6]
```diff
@@ -114,13 +114,13 @@ private class Settings(context: Context) {
         Context.MODE_PRIVATE
     )
 
-    var isTestMode: Boolean?
+    var isPreviewMode: Boolean?
         get() =
-            if (prefs.contains(IS_TEST_MODE))
-                prefs.getBoolean(IS_TEST_MODE, false)
+            if (prefs.contains(IS_PREVIEW_MODE))
+                prefs.getBoolean(IS_PREVIEW_MODE, false)
```
Comment (corycaywood) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/AppSettings.kt:120]: The default here should also be `true`.

File: testapp/src/main/res/layout/settings_menu_activity.xml [hunk 7]
```diff
@@ -189,11 +189,11 @@
...
 
         <com.google.android.material.switchmaterial.SwitchMaterial
-            android:id="@+id/switchTestMode"
+            android:id="@+id/switchPreviewMode"
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
             android:layout_marginTop="@dimen/small_12"
-            android:text="@string/lb_test_mode"
+            android:text="@string/lb_preview_mode"
             android:visibility="gone" />
```
Comment (corycaywood) [inline testapp/src/main/res/layout/settings_menu_activity.xml:197]: We can make this switch visible now.

Also, I think it needs to be updated to match the styling of the buttons. Maybe it just needs to be wrapped in `RelativeLayout` and add the horizontal divider beneath?

**Review by corycaywood — COMMENTED** [review]
Looking good 👍  See my comments below.

File: testapp/build.gradle (hunk 4 above)
Comment (khairul-alam-licon) [reply testapp/build.gradle:38]: Ah, yes I missed this spec.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniappSdkInitializer.kt (hunk 3 above)
Comment (khairul-alam-licon) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniappSdkInitializer.kt]: I have updated this part.

File: testapp/src/main/res/layout/settings_menu_activity.xml (hunk 7 above)
Comment (khairul-alam-licon) [reply testapp/src/main/res/layout/settings_menu_activity.xml]: Yeah, the Preview switch is visible now.

**Review by corycaywood — APPROVED** [review]
Looks good! 👍

**Pull request merged**